### ISSUE TITLE
AWS GovCloud Support

### ISSUE THREAD

**Issue by jbust23** [serverless/serverless#3748]
# This is a Bug Report

## Description

For bug reports:

* What went wrong?
Failed to deploy application on AWS GovCloud. 

* What did you expect should have happened?
Application should have successfully deployed on AWS GovCloud. The same service deploys successfully on non AWS GovCloud accounts. 

* What was the config you used?
```
service: gov-test-service

provider:
    name: aws
    runtime: python2.7
    region: us-gov-west-1
    
functions:
  hello:
    handler: handler.hello
```

* What stacktrace or error message from your provider did you see?
`TemplateURL must reference a valid S3 object to which you have access.`

For feature proposals:
* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
* If there is additional config how would it look

Similar or dependent issues:
N/A

## Additional Data

* ***Serverless Framework Version you're using***:
1.14.0
* ***Operating System***:
OSX 10.12.5
* ***Stack Trace***:
* ***Provider Error messages***:
`TemplateURL must reference a valid S3 object to which you have access.`

**Comment by JBust23** [comment]
I have found a solution to this. I looked into the `lib/plugins/aws/lib/updateStack.js` file and found that the `templateURL` variable has the https://s3.amazon.com URL hard coded. The problem was fixed by changing the URL to `https://s3-us-gov-west-1.amazonaws.com`

It would be ideal to have a permanent solution in a future release where we can switch between a public AWS account and a GovCloud account.
